### PR TITLE
Fix PHPDoc blocks

### PR DESCRIPTION
**Pull request by szepeviktor**
**Description**
1. Inserted `string` in all `@params` without type
2. Ran PHPStan `vendor/bin/phpstan analyse -l 2 src/`
3. Fixed everything I could

**Fixed issue**
#280

@acampos1916

### CODE REVIEW

**Comment by szepeviktor** [comment]
That `Config|ConfigInterface` thing needs to be cleaned up.

File: src/Adyen/Client.php [hunk 1]
```diff
@@ -8,6 +8,9 @@
 use Monolog\Logger;
 use Monolog\Handler\StreamHandler;
 
+/**
+ * @template TConfig of ConfigInterface
```
Comment (szepeviktor) [inline src/Adyen/Client.php:12]: This is [PHP generics](https://phpstan.org/blog/generics-in-php-using-phpdocs).

File: src/Adyen/Config.php [hunk 2]
```diff
@@ -67,15 +67,15 @@ public function set($key, $value)
     }
 
     /**
-     * @return mixed|null
+     * @return mixed
```
Comment (szepeviktor) [inline src/Adyen/Config.php:70]: There is no such a type as `mixed|anything`. `mixed` combines all types.

File: src/Adyen/Service/AbstractResource.php [hunk 3]
```diff
@@ -205,15 +204,11 @@ private function handleApplicationInfoInRequestPOS(array $params)
...
      * @return bool
      */
     private function isBase64Encoded($data)
     {
-        if (preg_match('%^[a-zA-Z0-9/+]*={0,2}$%', $data) && !empty($data)) {
-            return true;
-        } else {
-            return false;
-        }
+        return preg_match('%^[a-zA-Z0-9/+]*={0,2}$%', $data) && !empty($data);
```
Comment (szepeviktor) [inline src/Adyen/Service/AbstractResource.php:212]: I don't tell anyone about it 😉

**Review by cyattilakiss — APPROVED** [review]

**Review by peterojo — APPROVED** [review]

**Comment by AlexandrosMor** [comment]
Hello @szepeviktor,

We would like to thank you taking the time to open this pull request. This issue was solved in the later pull requests so for that reason we are going to close this pr.

kind regards,
Alexandros
Adyen

**Pull request closed without merging**